### PR TITLE
add a function to keep track of the user's token in state

### LINKED ISSUE #33: Unify the way token in handled across components
We should add a function here that passes the token value to the `App` component:
```props.setToken(token);```
I had to do it in our feature to make sure `App` starts fetching data from firebase right away if that makes sense.

_Originally posted by @dariatsvetkova in https://github.com/the-collab-lab/tcl-22-smart-shopping-list/pull/25#discussion_r618915306_

### PR DESCRIPTION
**Pull request by dariatsvetkova**
_For an example of how to fill this template out, [see this Pull Request](https://github.com/the-collab-lab/tcl-3-smart-shopping-list/pull/44)._ 

## Description

This unifies the way user token is handled across components. As soon as a new token is created or added by the user, it triggers a re-render of the `App` component, which retrieves the corresponding grocery list from Firebase.

## Related Issue

Closes #33 

## Acceptance Criteria

Starts displaying the grocery list and adding items to it as soon the user enters a token.

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|  ✓  | :bug: Bug fix              |
|   | :sparkles: New feature     |
|    | :hammer: Refactoring       |
|    | :100: Add tests            |
|    | :link: Update dependencies |
|    | :scroll: Docs              |

### Before

React App (2)

### After

React App (3)

## Testing Steps / QA Criteria

To test the feature, use one of the following ways:

1) Follow the link to see the preview: https://tcl-22-shopping-list--pr34-dt-token-handling-7rks31hk.web.app/
2) Launch the app locally from your terminal:
* Pull down this branch with `git pull origin dt-token-handling` and check that branch out with `git checkout origin dt-token-handling`
* Run `npm start`  to launch the app in your browser at [http://localhost:3000](http://localhost:3000)

When on application:

* If you don't have an existing grocery list, create one
* Open the local storage to see your current token; copy it
* Clear your local storage to remove the existing token
* Reload the app to see the Welcome screen
* Add your copied token
* You should see your list now and be able to add new items to it.

### CODE REVIEW

File: src/ShareToken.js [hunk 1]
```diff
@@ -22,6 +22,7 @@ const ShareToken = (props) => {
             variant: 'success',
           });
           localStorage.setItem('userToken', shareToken);
+          props.setToken(shareToken);
           history.push('/list');
```
Comment (kkatoch89) [inline src/ShareToken.js:26]: Love the fact that you're standardizing and bringing the token/update token code to one component. I noticed that in both the `Welcome` and `ShareToken` components, we seem to have `localStorage.setItem('userToken', shareToken/token)` in the same context. A chance for some refactoring perhaps?
Something like this in `App` would do the trick and we could pass it through the same props you're using to pass `setToken` to:
``` 
const updateToken = (newToken) => {
    setToken(newToken);
    localStorage.setItem('userToken', token);
  };
```

**Review by kkatoch89 — APPROVED** [review]
Love it! Nicely done @dariatsvetkova 🔥. I put in a suggestion for potential refactoring if you feel it makes sense.

File: src/ShareToken.js (hunk 1 above)
Comment (ablizben) [reply src/ShareToken.js:26]: Taking the advice given and asking questions on PRs - why am I on a team of such rockstars?

**Review by ablizben — APPROVED** [review]

**Comment by laurenmbeatty** [comment]
I'm going to throw this out here, even though it's not part of this PR...I don't think we need the `history.push` in either the `ShareToken` or `Welcome` components. That's why we have the `Redirect`. If there is a token set, it will redirect to the List page. I'm 👍  on this change, and maybe try removing the history.push in both of those functions and see what happens. I just tested it out with creating a brand new list and on trying to re-enter at the home page when I had an existing token. If I created a token, or had an existing token, it re-directed as expected.

**Review by laurenmbeatty — APPROVED** [review]
LGTM 👍.

**Review by cafloyd — APPROVED** [review]

File: src/ShareToken.js (hunk 1 above)
Comment (dariatsvetkova) [reply src/ShareToken.js:26]: Yes, @kkatoch89 that makes a lot of sense! I'll do that.

**Comment by dariatsvetkova** [comment]
@laurenmbeatty you are right, now that we have the value of the token in state, we don't need `history` anymore.

**Comment by dariatsvetkova** [comment]
Implemented @kkatoch89 's and @laurenmbeatty 's suggestions.

**Review by kkatoch89 — APPROVED** [review]
Looks 💯 to me! Thank you @dariatsvetkova!

**Comment by kkatoch89** [comment]
Lol sorry my bad, was trying to exit the GitHub app on my phone

**Review by laurenmbeatty — APPROVED** [review]
Great! Works when creating a brand new token and sharing a token. 🎉

**Pull request merged**